packages/snowpack/src/index.js

```diff
@@ -2,65 +2,62 @@ import { compareSignatures, isComponent } from '@prefresh/utils';
 
-          const compareSignaturesForPrefreshment = ${compareSignatures.toString()};
-          const shouldPrefreshBind = ${isComponent.toString()}
+          let prevRefreshReg;
+          let prevRefreshSig;
+          const module = {};
+          const compareSignatures = ${compareSignatures.toString()};
 
-          const __module_exports__ = []
+          if (import.meta.hot) {
+            prevRefreshReg = self.$RefreshReg$ || (() => {});
+            prevRefreshSig = self.$RefreshSig$ || (() => {});
 
-          const prevRefreshReg = self.$RefreshReg$ || (() => {});
-          const prevRefreshSig = self.$RefreshSig$ || (() => {});
-
-          self.$RefreshSig$ = () => {
-            let status = 'begin';
-            let savedType;
-            return (type, key, forceReset, getCustomHooks) => {
-              if (!savedType) savedType = type;
-              status = self.__PREFRESH__.sign(type || savedType, key, forceReset, getCustomHooks, status);
-              return type;
+            self.$RefreshReg$ = (type, id) => {
+              module[type.name] = type;
             };
-          };
 
-          self.$RefreshReg$ = (type, id) => {
-            __module_exports__.push(type.name);
-          };
+            self.$RefreshSig$ = () => {
+              let status = 'begin';
+              let savedType;
+              return (type, key, forceReset, getCustomHooks) => {
+                if (!savedType) savedType = type;
+                status = self.__PREFRESH__.sign(type || savedType, key, forceReset, getCustomHooks, status);
+                return type;
+              };
+            };
+          }
 
           ${contents}
 
-          self.$RefreshSig$ = prevRefreshSig;
-          self.$RefreshReg$ = prevRefreshReg;
-
-          if (import.meta.hot && __module_exports__.some(shouldPrefreshBind)) {
-            import.meta.hot.accept(({ module }) => {
+          if (import.meta.hot) {
+            self.$RefreshSig$ = prevRefreshSig;
+            self.$RefreshReg$ = prevRefreshReg;
+            import.meta.hot.accept((m) => {
               try {
-                for (let i in module) {
-                  if (typeof module[i] === 'function') {
-                    if (i in $CurrentModule$) {
-                      // We could add a check here on i.name if it's a component.
-                      compareSignaturesForPrefreshment(
-                        $CurrentModule$[i],
-                        module[i]
-                      );
-                    }
+                for (let i in m) {
+                  if (i === 'default') {
+                    const keyword = m[i].name;
+                    compareSignatures(module[keyword], m[i]);
+                  } else {
+                    compareSignatures(module[i], m[i]);
                   }
                 }
-                $CurrentModule$ = module;
               } catch(e) {
                 import.meta.hot.invalidate();
+                self.location.reload();
```

Comment:
Using the Vite plugin as an example this was my attempt at a) using the example provided, and b) updating the Snowpack plugin to reflect Vite (so it’s easier to maintain). Is this correct?

Reply:
Afaik `import.meta.hot.invalidate()` does `self.location.reload()` in snowpack

Reply:
Good catch! You’re right.